python/pyiceberg/catalog/base.py

```diff
@@ -138,7 +138,7 @@ def rename_table(self, from_identifier: str | Identifier, to_identifier: str | I
             Table: the updated table instance with its metadata
 
         Raises:
-            TableNotFoundError: If a table with the name does not exist
+            NoSuchTableError: If a table with the name does not exist
```

Comment:
Can you open a separate PR for these refactoring changes?

Reply:
Sure thing, I've opened https://github.com/apache/iceberg/pull/5421

Reply:
Merged! Thanks